src/psyclone/psyir/nodes/call.py

```diff
@@ -355,6 +355,14 @@ def is_available_on_device(self):
         '''
         return False
 
+    @property
+    def datatype(self):
+        '''
+        :raises NotImplementedError: TODO #1799
+        '''
+        raise NotImplementedError("datatype property not yet implemented for "
+                                  "Call nodess")
+
```

Comment:
I think this should be one level up, in the DataNode class, as all DataNodes will have datatypes.

And again, I am wondering it it will be easier to return DeferredType instead of error because we keep inside the type system semantics, and for example:
`a == call(b) + c` we could end with a bool even if a, b, c are DeferredType/UnknwonType without dealing with any errors.

Reply:
Ooh, nice. Done.